Work with a DAG

milljm opened the pull request:
Use a DAG object to control what gets built.

Commit:
Work with a DAG

Use a DAG object to control what gets built.

Commit:
Upgrade to PETSc 3.7.6 as default

Upgrade to latest petsc, beautify the printed output
a bit.

moosebuild commented:
Job [Package Builder](https://www.moosebuild.org/job/104550/) on bb2751a : canceled by @milljm

ITS WRONG

Commit:
Fix issue with subprocess not returning

Interesting issue with thread locking and the subprocess.Popen
object not immediately returning.

Allow jobs that require almost no processing power to 'just go'.

moosebuild commented:
Job [Package Builder](https://www.moosebuild.org/job/104641/) on d89927f : invalidated by @milljm

Can&#39;t open profile/psizeof_f08.f90 for writing at ../../../../../ompi/mpi/fortran/base/gen-mpi-sizeof.pl line 187. make[2]: *** [profile/psizeof_f08.f90] Error 2 make[2]: Leaving directory `/tmp/moose_package_build_temp/openmpi-clang/openmpi-1.10.2/clang-opt/ompi/mpi/fortran/use-mpi-f08&#39; make[1]: *** [all-recursive] Error 1 make[1]: Leaving directory `/tmp/moose_package_build_temp/openmpi-clang/openmpi-1.10.2/clang-opt/ompi&#39; make: *** [all-recursive] Error 1 make: INTERNAL: Exiting with 25 jobserver tokens available; should be 24! Failed to make  Error while working on: openmpi-clang

Commit:
OpenMPI does not seem to build with load balancing

moosebuild commented:
Job [Package Builder](https://www.moosebuild.org/job/104694/) on 4ce1764 : invalidated by @milljm

cleaned up pip cache, causing issues during install

Commit:
Prevent pip from using cache

Modify active list print out.
Change logic for queueing jobs.

milljm commented:
Errors present were due to disk space issues...

moosebuild commented:
Job [Test Mpich Clang petsc default vtk](https://www.moosebuild.org/job/104821/) on 7653284 : invalidated by @milljm

out of space :(

moosebuild commented:
Job [Trilinos mpich gcc petsc default vtk](https://www.moosebuild.org/job/104822/) on 7653284 : invalidated by @milljm

out of space :(
